### PR TITLE
#9323: Introduce template for new ttnn pull requests

### PR DESCRIPTION
**Pull request by cfjchu**
### Ticket
#9323 

### Problem description
We would like to improve pull-requests created in TTNN so that reviewers have more context for the changes involved in a pull-request.

### What's changed
This introduces a template for future pull-requests opened under `ttnn`. These changes are introduced to begin defining  requisite information that we'd like to see when new pull-requests are created in TTNN. The intent is to better define the problem, the solution and impact of changes.

### Checklist
- [x] Post commit CI passes: N/A
- [x] Model regression CI testing passes (if applicable): N/A
- [x] New/Existing tests provide coverage for changes: N/A

### CODE REVIEW

**Review by ayerofieiev-tt — APPROVED** [review]

**Review by arakhmati — APPROVED** [review]

**Review by bbradelTT — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE/ttnn.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+## Related Issues
+
+Please link related issues and describe the problem this pull-request is trying to resolve.
+
+## Description
+
+Please include a summary of the changes and the related issues. This template is for changes in the `ttnn` subdirectory.
+- Describe *what* changed in the pull-request
+- Describe *why* these changes were introduced
```
Comment (cfjchu) [inline .github/PULL_REQUEST_TEMPLATE/ttnn.md:9]: @eyonland we can discuss in our team-sync whether this should really be contained within the issue itself. It sounds like maybe we should have a line here that describes *how* the changes in the pull-request solve the problem defined in the issue

Comment (ayerofieiev-tt) [reply .github/PULL_REQUEST_TEMPLATE/ttnn.md:9]: 2c:
It should be within the issue so the one who works on the issue has The Context.
It should be within the PR so those reading are also in The Context.

**Comment by tt-rkim** [comment]
Looping in @TT-BrianLiu as well

How often do we see changes in tt-lib? Should we start enforcing static-check level gating to ensure it doesn't change?

**Review by tt-rkim — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE/ttnn.md (hunk 1 above)
Comment (eyonland) [reply .github/PULL_REQUEST_TEMPLATE/ttnn.md:9]: One format I've seen in the past is something like you see below.   This does restate the problem from the issue briefly, but also has the solution for what was done.

#Ticket Number: similar title to what is on the issue

Problem:
- Describe the problem
- Another bullet about the problem

Solution:
- What was changed
- another bullet about what was changed

File: .github/PULL_REQUEST_TEMPLATE/ttnn.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+- Describe implications of these code-changes: e.g. runtime/compile-time perf impact, usability, future code-changes
+
+## Checklist
+
+- [ ] Post commit CI passes.
+- [ ] Related model testing
+- [ ] New tests added
+- [ ] Change is made to ttnn and not tt_lib :)
```
Comment (eyonland) [inline .github/PULL_REQUEST_TEMPLATE/ttnn.md:17]: I have mixed feelings about auto generated lists.   When they do not apply you have to explicitly delete them.   That is good when you want to drive behavior towards best practices.  (Maybe this is a good way...)     Let's talk more about this as well.

Comment (cfjchu) [reply .github/PULL_REQUEST_TEMPLATE/ttnn.md]: Discussed in our team-sync. I think that's the right way to look at it; this should serve as reminders to help drive developers towards best practices.

File: .github/PULL_REQUEST_TEMPLATE/ttnn.md (hunk 1 above)
Comment (cfjchu) [reply .github/PULL_REQUEST_TEMPLATE/ttnn.md]: Thanks for the feedback, everyone. I've modified the template based on discussions we've had.

**Pull request merged**